tram_fsm/lib/tram_fsm.ex

```diff
@@ -97,11 +130,21 @@ defmodule TramFsm do
     apply_transition(transition, transition_name, old_state)
   end
 
+  @doc """
+  Send error for invalid transition application.
+  """
+  @spec apply_transition(nil, atom(), tram_state()) ::
```

Comment:
Лучше одну спеку на всех, наподобие
```suggestion
  @spec apply_transition(nil | transition(), atom(), tram_state()) :: 
    transition_application_error() | transition_application_success()
```

Reply:
добавил